src/fondant/executor.py

```diff
@@ -300,6 +289,32 @@ def execute(self, component_cls: t.Type[Component]) -> None:
 
         self.upload_manifest(output_manifest, save_path=self.output_manifest_path)
 
+    def _upload_manifest_reference(
```

Comment:
the naming in this function is a bit confusing cache key vs manifest reference

Reply:
Good call, i'll change it